### PR TITLE
Add Order minion

### PR DESCRIPTION
**Pull request by eclarizio**
This PR adds what used to be the "update order listener" service from catalog-api into a minion. A subsequent PR needs to be created in the catalog-api repo to remove that service so that this one becomes the standard instead.

When writing the specs for the minion I realized we weren't even using the `spec_helper.rb` file, so I included that in the `.rspec` file, which promptly broke things because `.use_transactional_fixtures` is an active record thing and we're not using active record in this project.

It also utilizes the new endpoint created for order in https://github.com/ManageIQ/catalog-api/pull/326, so could technically be merged into this repo but not used without breaking until that PR has been merged.

@miq-bot add_reviewer @lindgrenj6, @syncrou

### CODE REVIEW

File: bin/catalog-api-order-minion [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+STDOUT.sync = true
+
+require "bundler/setup"
+require "catalog/minion"
+
+def parse_args
+  require 'optimist'
+  opts = Optimist.options do
+    opt :queue_host, "Hostname of the Platform's kafka queue", :type => :string,
```
Comment (lindgrenj6) [inline bin/catalog-api-order-minion:14]: More of just a general question, how are the Optimist options used? Is it GNU style (`--queue_host`) or just `QUEUE_HOST` as an env var?

**Review by lindgrenj6 — APPROVED** [review]
Looks good to me, I'm assuming you're going to remove the order minion from `catalog-api` in a new PR?

File: bin/catalog-api-order-minion (hunk 1 above)
Comment (syncrou) [reply bin/catalog-api-order-minion:14]: @lindgrenj6 They're environment variables. You can see them set up here: https://console.insights-dev.openshift.com/console/project/db-catalog/browse/dc/catalog-api-minion?tab=environment

**Review by syncrou — COMMENTED** [review]
This looks good, I still need to complete my testing as `persist_ref` was causing me issues. I'm assuming it may here too. Once I complete the approval testing I can add more feedback here.

File: lib/catalog/api/minion/order.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        end
+
+        private
+
+        def post_internal_notify(payload, payload_params)
+          RestClient.post(internal_notify_url(payload['task_id']), payload_params, identity_headers(persist_ref))
+        end
+
+        def internal_notify_url(task_id)
+          config = ::CatalogApiClient.configure
```
Comment (syncrou) [inline lib/catalog/api/minion/order.rb:41]: See my changes here: https://github.com/ManageIQ/catalog-api-minion/pull/10/files#diff-68e153a4bcfe0f001ff02ef7f54a836b which are required to statically define the catalog-ci url until we can get an e2e-deploy repo setup which will dynamically override `config = ::CatalogApiClient.configure` to have `config.host` return the correct host and `config.port` return the correct port.

File: spec/catalog/api/minion/order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      described_class.new("localhost", "9092")
+    end
+    let(:message) { ManageIQ::Messaging::ReceivedMessage.new(nil, "request_started", payload, nil, nil, nil) }
+    let(:payload) { {"task_id" => "123"} }
+    let(:payload_params) do
+      {:payload =>  message.payload, :message => message.message}
+    end
+    let(:config) do
+      test_config = ::CatalogApiClient.configure
+      test_config.host = "localhost:3000"
```
Comment (syncrou) [inline spec/catalog/api/minion/order_spec.rb]: Will need changes here as well with the static url set in `order.rb`

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -6,11 +6,12 @@
 ENV["RAILS_ENV"] ||= "test"
 raise "Specs must be run in test environment" if ENV["RAILS_ENV"] != "test"
 
+require "webmock/rspec"
 require "catalog/api/minion/logging"
 Catalog::Api::Minion.logger = Logger.new("/dev/null")
 
 RSpec.configure do |config|
-  config.use_transactional_fixtures = true
+  # config.use_transactional_fixtures = true
```
Comment (syncrou) [inline spec/spec_helper.rb:14]: Didn't know about the `ActiveRecord` requirement here, but makes sense.

**Review by syncrou — COMMENTED** [review]
It looks like it just needs the changes for the static URL.  We'll also want the PR pulling out the Order Threaded listener so we can simultaneously add these changes and pull that out of catalog.

**Review by syncrou — APPROVED** [review]
LGTM 👍

**Review by syncrou — APPROVED** [review]
LGTM 👍

**Pull request merged**